scripts/lint-style.lean

```diff
@@ -12,10 +13,58 @@ import Cli.Basic
 
 This files defines the `lint-style` executable which runs all text-based style linters.
 The linters themselves are defined in `Mathlib.Tactic.Linter.TextBased`.
+
+In addition, this checks that `Mathlib.Init` is (transitively) imported in all of mathlib.
 -/
 
 open Cli Mathlib.Linter.TextBased
 
+/-- Check that `Mathlib.Init` is transitively imported in all of Mathlib. -/
+def checkInitImports : IO Bool := do
+  -- Find any file in the Mathlib directory which does not contain any Mathlib import.
+  -- The current check (for any line starting with "import Mathlib") is not perfect
+  -- (for instance, it would detect a multi-line comment containing the line import Mathlib),
+  -- but is good enough in practice.
+  let mut modulesWithoutMathlibImports := #[]
+  -- We simply parse `Mathlib.lean`; this removes the need for external tools like grep.
+  let allModules := (← IO.FS.lines "Mathlib.lean").map (·.stripPrefix "import ")
+    |>.filter (·.startsWith "Mathlib")
+  for module in allModules do
+    let path := (System.mkFilePath (module.split (· == '.'))).addExtension "lean"
+    let hasMathlibImport := (← IO.FS.lines path).any fun s ↦ s.startsWith "import Mathlib"
+    if !hasMathlibImport then
+      modulesWithoutMathlibImports := modulesWithoutMathlibImports.push module
+
+  -- We check that each of these is imported in `Mathlib/Init.lean`.
+  let initImports := (← IO.FS.lines "Mathlib/Init.lean").filter (·.startsWith "import ")
+    |>.map (·.stripPrefix "import ")
+  let missing := modulesWithoutMathlibImports.filter (fun mod ↦ !initImports.contains mod)
+    -- `DeclarationNames` is imported transitively in `Mathlib/Init.lean`.
+    |>.erase "Mathlib.Tactic.DeclarationNames"
+  if missing.size > 0 then
+    IO.eprintln s!"error: the following {missing.size} module(s) do not import Mathlib.Init: \
+      {missing}"
+    return True
```

Comment:
```suggestion
    return true
```
I think that you should be returning a `Bool`, not a `Prop`, right?

Similar comment for the later return values.

Reply:
Oh yes! Fixed.